src/shared/components/palette/collection/write-collection-field.component.ts

```diff
@@ -48,15 +48,15 @@ export class WriteCollectionFieldComponent extends AbstractFieldWriteComponent i
   }
 
   buildCaseField(item, index: number): CaseField {
-    return {
+    return Object.assign(new CaseField(),{
```

Comment:
just to improve readability, can you create a private method in this class and call it `newCaseField` and then do the 
`Object.assign(new CaseField()` there?

Reply:
ok